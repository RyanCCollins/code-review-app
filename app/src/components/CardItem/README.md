## CardItem Component
A component that ...

### Example

```js
<CardItem />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
