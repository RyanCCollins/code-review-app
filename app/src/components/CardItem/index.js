import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';

const CardItem = ({
  children,
}) => (
  <div className={styles.cardItem}>
    {children}
  </div>
);

CardItem.propTypes = {
  children: PropTypes.node.isRequired,
};

export default cssModules(CardItem, styles);
