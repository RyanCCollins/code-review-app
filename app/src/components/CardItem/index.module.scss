.cardItem {
  padding: 16px;
  background: #FFF;
  margin-bottom: 14px;
  -webkit-border-radius: 1px;
  -moz-border-radius: 1px;
  border-radius: 1px;
  border: 1px solid #E0E0E0;
  box-shadow: 0 1px 2px #E0E0E0;
}
