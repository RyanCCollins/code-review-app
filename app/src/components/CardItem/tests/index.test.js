import CardItem from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<CardItem />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <CardItem>
        <div>
          Hello World
        </div>
      </CardItem>
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
