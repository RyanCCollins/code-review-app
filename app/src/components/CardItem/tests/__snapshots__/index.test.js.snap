exports[`<CardItem /> should render with default props 1`] = `
<div>
  <div>
    Hello World
  </div>
</div>
`;
